arch/riscv/core/pmp.c

```diff
@@ -493,17 +529,19 @@ static inline unsigned int z_riscv_pmp_thread_init(unsigned long *pmp_addr,
 }
 #endif
 
-#ifdef CONFIG_PMP_STACK_GUARD
+#if (defined(CONFIG_PMP_STACK_GUARD) || defined(CONFIG_MEM_ATTR))
 
 #ifdef CONFIG_MULTITHREADING
 /**
  * @brief Prepare the PMP stackguard content for given thread.
  *
  * This is called once during new thread creation.
  */
-void z_riscv_pmp_stackguard_prepare(struct k_thread *thread)
+void z_riscv_pmp_kernelmode_prepare(struct k_thread *thread)
 {
 	unsigned int index = z_riscv_pmp_thread_init(PMP_M_MODE(thread));
+
```

Comment:
If user-defined PMP entries can’t be global when user mode is enabled, we need to configure user-defined entries for supervisor thread.